framework/resource/metricsresource/crud_resource_ops_wrapper.go

```diff
@@ -0,0 +1,178 @@
+package metricsresource
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/framework/resource/internal"
+)
+
+type crudResourceOpsWrapperConfig struct {
+	Ops framework.CRUDResourceOps
+
+	Name string
+}
+
+type crudResourceWrapperOps struct {
+	underlying framework.CRUDResourceOps
+
+	name string
+}
+
+func newCRUDResourceWrapperOps(config crudResourceOpsWrapperConfig) (*crudResourceWrapperOps, error) {
+	if config.Ops == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Ops must not be empty")
+	}
+
+	if config.Name == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.Name must not be empty")
+	}
+
+	o := &crudResourceWrapperOps{
+		underlying: config.Ops,
+
+		name: toCamelCase(config.Name),
+	}
+
+	return o, nil
+}
+
+func (o *crudResourceWrapperOps) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	sl := o.name
+	rl := o.underlying.Name()
+	ol := "GetCurrentState"
+
+	operationCounter.WithLabelValues(sl, rl, ol).Inc()
+
+	t := prometheus.NewTimer(operationHistogram.WithLabelValues(sl, rl, ol))
+	defer t.ObserveDuration()
+
+	v, err := o.underlying.GetCurrentState(ctx, obj)
+	if err != nil {
+		operationErrorCounter.WithLabelValues(sl, rl, ol).Inc()
```

Comment:
I guess this is just how it is in the existing imoplementations but I feel like this never really worked out well. I feel like we should set the error counter to zero in case no error occurs and in case of an error we should set it to one. I would need to play around with the dashboard magic a bit. Uncertain. /cc @JosephSalisbury

Reply:
A counter makes more sense to me here, the issue is the sample size is fairly low - this might be a candidate for log level alerting.